include/E57Simple.h

```diff
@@ -0,0 +1,888 @@
+
+      uint8_t *colorRed{ nullptr };      //!< pointer to a buffer with the Red color coefficient. Unit is unspecified
+      uint8_t *colorGreen{ nullptr };    //!< pointer to a buffer with the Green color coefficient. Unit is unspecified
+      uint8_t *colorBlue{ nullptr };     //!< pointer to a buffer with the Blue color coefficient. Unit is unspecified
+      int8_t *isColorInvalid{ nullptr }; //!< Value = 0 if the color is considered valid, 1 otherwise
+
+      float *sphericalRange{
+         nullptr
+      }; //!< pointer to a buffer with the range (in meters) of points in spherical coordinates. Shall be non-negative
+      float *sphericalAzimuth{
+         nullptr
+      }; //!< pointer to a buffer with the Azimuth angle (in radians) of point in spherical coordinates
+      float *sphericalElevation{
+         nullptr
+      }; //!< pointer to a buffer with the Elevation angle (in radians) of point in spherical coordinates
+      int8_t *sphericalInvalidState{ nullptr }; //!< Value = 0 if the range is considered valid, 1 otherwise
+
+      int32_t *rowIndex{ nullptr }; //!< pointer to a buffer with the row number of point (zero based). This is useful
+                                    //!< for data that is stored in a regular grid.  Shall be in the interval (0, 2^31).
+      int32_t *columnIndex{
+         nullptr
+      }; //!< pointer to a buffer with the column number of point (zero based). This is useful for data that is stored
+         //!< in a regular grid. Shall be in the interval (0, 2^31).
+      int8_t *returnIndex{
+         nullptr
+      }; //!< pointer to a buffer with the number of this return (zero based). That is, 0 is the first return, 1 is the
+         //!< second, and so on. Shall be in the interval (0, returnCount). Only for multi-return sensors.
+      int8_t *returnCount{
+         nullptr
+      }; //!< pointer to a buffer with the total number of returns for the pulse that this corresponds to. Shall be in
+         //!< the interval (0, 2^7). Only for multi-return sensors.
+
+      double *timeStamp{
+         nullptr
+      }; //!< pointer to a buffer with the time (in seconds) since the start time for the data, which is given by
+         //!< acquisitionStart in the parent Data3D Structure. Shall be non-negative
+      int8_t *isTimeStampInvalid{ nullptr }; //!< Value = 0 if the timeStamp is considered valid, 1 otherwise
+
+      // E57_EXT_surface_normals
+      float *normalX{ nullptr }; //!< The X component of a surface normal vector (E57_EXT_surface_normals).
+      float *normalY{ nullptr }; //!< The Y component of a surface normal vector (E57_EXT_surface_normals).
+      float *normalZ{ nullptr }; //!< The Z component of a surface normal vector (E57_EXT_surface_normals).
+
+      bool ( *pointDataExtension )( ImageFile imf, StructureNode proto, std::vector<SourceDestBuffer> &destBuffers ){
```

Comment:
I'm not sure whether these extension methods make sense in E57Simple API at all. One can use the current API when writing custom fields or doing anything more complex.

If kept, we should probably consider using std::function.

Reply:
I would agree with removing them.